extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/Util.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.binding.onewire.internal;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.measure.quantity.Dimensionless;
+import javax.measure.quantity.Temperature;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.library.dimension.Density;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.types.State;
+
+/**
+ * The {@link Util} is a set of helper functions
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+public class Util {
+
+    /**
+     * convert a number to a list of booleans
+     *
+     * @param decimal
+     * @return
+     */
+    public static List<Boolean> decimalTypeToBooleanList(DecimalType decimal) {
+        List<Boolean> returnValues = new ArrayList<Boolean>();
+        int value = decimal.intValue();
+        for (int i = 0; i < 8; i++) {
+            returnValues.add((value & (1 << i)) > 0);
+        }
+        return returnValues;
+    }
+
+    /**
+     * convert measured value to temperature with offset
+     *
+     * @param temperature measured value
+     * @param offset offset to apply
+     * @return originalState if not DecimalType, temperature otherwise
+     * @throws OwException in case of conversion error
+     */
+    public static QuantityType<Temperature> convertTemperature(QuantityType<Temperature> temperature,
```

Comment:
The offset handling should in the future be done by the new Offset-Profile. Please see #5679 which provides a system profile which can be configured for any channel. Therefor I suggest you drop the binding specific offset support here.

Reply:
Do you have an ETA for merging? I'll remove it as soon as this is merged.

Reply:
@J-N-K we are about to merge #5679. You can remove the binding specific offset handling.